### PR TITLE
Make server transmit Error packet if client request is not read or write

### PR DESCRIPTION
**Pull request by jasLogic**
This fixes https://github.com/connorkuehl/tftp/issues/19.

### CODE REVIEW

File: src/server.rs [hunk 1]
```diff
@@ -43,16 +43,13 @@ impl Server {
...
-            Some(Direction::Put(wq))
+            Direction::Put(wq)
         } else {
-            None
-        };
-
-        let direction = match direction {
-            None => return Err(io::ErrorKind::InvalidInput.into()),
-            Some(d) => d,
+            let error = Packet::error(Code::IllegalOperation, "illegal operation");
```
Comment (connorkuehl) [inline src/server.rs:50]:
```suggestion
            let error = Packet::error(Code::IllegalOperation, format!("{}", Code::IllegalOperation));
```

The `Code` enum can be formatted into a String with its implementation of the `Display` trait.

**Review by connorkuehl — CHANGES_REQUESTED** [review]
Thanks for the PR, @jasLogic! I just have 1 comment inline.

File: src/server.rs (hunk 1 above)
Comment (jasLogic) [reply src/server.rs]: Done

**Review by connorkuehl — APPROVED** [review]

**Comment by connorkuehl** [comment]
Thank you 🙂

**Pull request merged**